ddtrace/internal/writer.py

```diff
@@ -140,9 +153,33 @@ def put(self, item):
                     idx = random.randrange(0, qsize)
                     self.queue[idx] = item
                     log.warning('Writer queue is full has more than %d traces, some traces will be lost', self.maxsize)
+                    self.dropped += 1
+                    self._update_stats(item)
```

Comment:
any reasonable way to consolidate into a `finally:` or something? Worried about branches that might forget to set this.

Reply:
Actually, yeah, `self._update_stats(item)` needs to be dedented no?

Reply:
Moving this elsewhere would make `self.mutex` being release and then reacquired. For performance reason, this might not be the best choice.

Reply:
Makes sense.

Reply:
oh, ignore my other comment, too early in the morning, we have `self.put(item)` to retry below.